query-engine/driver-adapters/executor/src/recording.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  type DriverAdapter,
+  type Query,
+  type Result,
+  type ResultSet,
+} from "@prisma/driver-adapter-utils";
+import { RetryHandler } from "undici";
+
+export function recording(adapter: DriverAdapter) {
+  const recordings = createInMemoryRecordings();
+
+  return {
+    recorder: recorder(adapter, recordings),
+    replayer: replayer(adapter, recordings),
+  };
+}
+
+function recorder(adapter: DriverAdapter, recordings) {
+  return {
+    provider: adapter.provider,
+    startTransaction: () => {
+      throw new Error("Not implemented");
+    },
+    getConnectionInfo: () => {
+      return adapter.getConnectionInfo!();
+    },
+    queryRaw: (params) => {
+      return adapter.queryRaw(params).then((result) => {
+        recordings.addQueryResults(params, result);
+        return result;
+      });
+    },
```

Comment:
Can we use a unified `async/await` syntax here, like done below in `replayer.queryRaw`?

```suggestion
    queryRaw: async (params) => {
      const result = await adapter.queryRaw(params);
      recordings.addQueryResults(params, result);
      return result;
    },
```